ocp_utilities/infra.py

```diff
@@ -81,6 +85,46 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes_with_conditions = {}
```

Comment:
please define as defaultdict

Reply:
done.
This is the dict that I plan to use
```
    default_unhealthy_node_condition_type = dict(
        DiskPressure=Node.Condition.Status.TRUE,
        MemoryPressure=Node.Condition.Status.TRUE,
        PIDPressure=Node.Condition.Status.TRUE,
        NetworkUnavailable=Node.Condition.Status.TRUE, 
    )
```

Reply:
default_healthy_node_condition_type = {"DiskPressure": Node.Condition.Status.FALSE,  ...}

Reply:
Done